src/sentry/middleware/env.py

```diff
@@ -9,9 +9,11 @@ class SentryEnvMiddleware(object):
     def process_request(self, request):
         # bind request to env
         env.request = request
+        env.tenant = None
```

Comment:
need to ensure celery tasks are wired to ensure tenant is cleared on each task run (I forget if it hits our request_finished signal)